lib/views/login/login_form.dart

```diff
@@ -77,32 +83,69 @@ class LoginFormState extends ConsumerState<LoginForm> {
     );
   }
 
+  @override
+  void dispose() {
+    _email.dispose();
+    _password.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Column(
-      mainAxisAlignment: MainAxisAlignment.center,
-      crossAxisAlignment: CrossAxisAlignment.stretch,
-      children: [
-        _emailTextField,
-        const SizedBox(height: 20),
-        Stack(
-          alignment: AlignmentDirectional.centerEnd,
-          children: [
-            _passwordTextField,
-            _forgetPasswordButton(context),
-          ],
-        ),
-        const SizedBox(height: 20),
-        _loginButton,
-      ],
+    return Form(
+      key: _loginFormKey,
+      autovalidateMode: _isStatedValidation
+          ? AutovalidateMode.onUserInteraction
+          : AutovalidateMode.disabled,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          _emailTextField,
+          const SizedBox(height: 20),
+          Stack(
+            alignment: AlignmentDirectional.centerEnd,
+            children: [
+              _passwordTextField,
+              _forgetPasswordButton(context),
+            ],
+          ),
+          const SizedBox(height: 20),
+          _loginButton,
+        ],
+      ),
     );
   }
 
   Future<void> _login() async {
+    setState(() {
```

Comment:
@nmint8m As we're using `river_pod` for state management, please put all the business logic into `ViewModel`  and avoid using `setState` 🙏